c7n/resources/support.py

```diff
@@ -22,3 +22,16 @@ class resource_type(TypeInfo):
     def get_client(self):
         region = get_support_region(self)
         return local_session(self.session_factory).client('support', region_name=region)
+
+
+
+@resources.register("aws.advisor-check")
+class AdvisorCheck(QueryResourceManager):
+    class resource_type(TypeInfo):
+        service = "support"
+        enum_spec = ('describe_trusted_advisor_checks', 'checks', None)
+        detail_spec = ('describe_trusted_advisor_check_result', 'checkId', 'id', 'result')
+        arn_type = "checks"
+        arn_service = "trustedadvisor"
+        name = id = "checkId"
```

Comment:
Should we gracefully handle the `SubscriptionRequiredException` error folks will get if they try this in an account without premium support? Otherwise they'll see:

```
botocore.exceptions.ClientError: An error occurred (SubscriptionRequiredException) when calling the DescribeTrustedAdvisorChecks operation: Amazon Web Services Premium Support Subscription is required to use this service.
```

We'll also need to pass `language` to the describe call. I think it's _probably_ safe to just assume `en` unless someone provides another query?

```suggestion
class DescribeAdvisorCheck(DescribeSource):
    def resources(self, query):
        if not query:
            query = {'language': 'en'}
        return super().resources(query)


@resources.register("aws.advisor-check")
class AdvisorCheck(QueryResourceManager):
    class resource_type(TypeInfo):
        service = "support"
        enum_spec = ('describe_trusted_advisor_checks', 'checks', None)
        detail_spec = ('describe_trusted_advisor_check_result', 'checkId', 'id', 'result')
        arn_type = "checks"
        arn_service = "trustedadvisor"
        name = id = "checkId"

    source_mapping = {
        "describe": DescribeAdvisorCheck,
    }
```

Reply:
applied